### ISSUE TITLE
Property accessors to unknown keys doesn't throw errors.

### ISSUE THREAD

**Issue by RIP21** [Microsoft/TypeScript#27950]
**Code**
```ts
const a = (e: { b: string }) => {
    e['a'] = "f" 
}
```

**Expected behavior:**
```ts
const a = (e: { b: string }) => {
    e['a'] = "f" // Errors out since there is no such field
}
```
**Actual behavior:**
```ts
const a = (e: { b: string }) => {
    e['a'] = "f" // Ignored
}
```

**Playground Link:** 
https://www.typescriptlang.org/play/index.html#src=const%20a%20%3D%20(e%3A%20%7B%20b%3A%20string%20%7D)%20%3D%3E%20%7B%0D%0A%20%20%20%20e%5B'a'%5D%20%3D%20%22f%22%20%2F%2F%20Errors%20out%20since%20there%20is%20no%20such%20field%0D%0A%7D%0D%0A

**Comment by andy-ms** [comment]
You need to enable `"noImplicitAny": true` (or better, `"strict": true`) in your `tsconfig.json` compilerOptions to get that kind of error.

**Comment by RIP21** [comment]
@andy-ms now I feel stupid. I think I had it strict on. Let me check.

**Comment by RIP21** [comment]
@andy-ms weird.
I have strict and all other checks in my `tsconfig.json`
But didn't get this error in my IDE. (IntelliJ)
I have TS 3.1.1 and this IDE just runs language server if I'm not wrong, and shows what server says in the editor. But server says nothing =/

**Comment by RIP21** [comment]
@andy-ms Same behavior for VS Code tho. What can be an issue?

**Comment by RIP21** [comment]
@andy-ms Same behaviour in codesandbox https://codesandbox.io/s/o56j3ynkrq

The only place where it actually errors out is in Playground.

**Comment by RIP21** [comment]
` "suppressImplicitAnyIndexErrors": true`
Key to success T_T

Sorry for spam.
